syntax = "proto3";
option optimize_for = LITE_RUNTIME;

//善举类型表
message KinddeedType {
    int64 id            = 1;    //善举类型序号，0,1,2,3..，唯一
    string desc         = 2;    //善举类型名称,超度逝者,事业顺达，学业有成，... ...等等
}

//善举表
message Kinddeed {
    string id           = 1; //善举编号，唯一
    string name         = 2; //善举名称
    string owner        = 3; //哪位法师或基金会成员添加的此善举
    int64  type         = 4; //善举类型
    string lasttime     = 5; //善举最后修改的时间
    bool   online       = 6; //是否已经是上线，0为下线状态，1为上线状态
}

//善举图片描述表
message KinddeedDetail {
    string kdid         = 1; //善举编号
    int64  sequence     = 2; //善举描述的序号0,1,2,3...
    string hash         = 3; //善举描述图片的hash
}

//善举规格表
//全局共同使用此一张表，再善举下线后，删除此表中的对应的善举id表项
message KinddeedSpec {
    string kdid         = 1;    //善举id
    int64  sequence     = 2;    //善举规格序号
    string desc         = 3;    //善举规格描述
    int64  price        = 4;    //价格
}

//善行评论标签表，全局统一
message CommentLabel {
    int64 id            = 1;    //标签id，0,1,2,3..， 唯一
    string desc         = 2;    //标签内容,处理及时，超越期望，... ...等等
}

//善举点评表，任何人都可以参与，包括购买者和非购买者
message BeforeComment {
    string owner        = 1;    //账户id，前二者联合唯一
    string kdid         = 2;    //善举id
    int64  satisfaction = 3;    //满意度,0很赞，1满意，2体验不佳
    string labels       = 4;    //善举体验标签数据
    string comment      = 5;    //评价
    string timestamp    = 6;    //评论生成时间
}

//订单表。
message Order {
    string id           = 1; //订单编号，唯一
    string owner        = 2; //订单所有者，哪个用户祈求了此善举订单
    string kdowner      = 3; //对应善举所有者
    string kdid         = 4; //对应的善举id
    int64  specid       = 5; //对应的善举规格id
    int64  count        = 6; //祈求的此善举的规格的个数
    int64  amount       = 7; //善举总价格
    string timestamp    = 8; //订单生成时间
}

//善举凭证表
message KinddeedProof {
    string orderid      = 1; //善举订单编号，对应Order表中的id，唯一
    string owner        = 2; //对应善举凭证所有者
    string proof        = 3; //善举hash，唯一
    string timestamp    = 4; //法师或寺院尽心善举时的时间
    bool   approved     = 5; //是否已经批准此善举凭证，0尚未批准，1已经批准
}

//善举动态评论表
message AfterComment {
    string orderid      = 1;    //善举凭证id,也就是订单id,唯一
    string owner        = 2;    //账户id
    string comment      = 3;    //评价
    string timestamp    = 4;    //评论时间
}

//善举所有者信用值表
message Credit {
    string id           = 1;    //善举所有者
    int64  value        = 2;    //信用值   
}

//善举所有者信用值排名表,只排名前100信用值棒
message CreditRanking {
    int64  ranking      = 1;    //名次 1,2,3,....100
    string id           = 2;    //善举所有者
    int64  value        = 3;    //信用值   
}

//购买者功德值表
message Merit {
    string id           = 1;    //购买者
    int64  value        = 2;    //功德值   
}

//购买者功德值排名表,只排名前100功德值棒
message MeritRanking {
    int64  ranking      = 1;    //名次 1,2,3,....100
    string id           = 2;    //购买者
    int64  value        = 3;    //功德值   
}

